quodlibet/quodlibet/ext/events/synchronizedlyrics.py

```diff
@@ -0,0 +1,305 @@
+        self.textview.set_editable(False)
+        self.textview.set_cursor_visible(False)
+        self.textview.set_wrap_mode(Gtk.WrapMode.WORD)
+        self.textview.set_justification(Gtk.Justification.CENTER)
+        self.scrolled_window.add_with_viewport(self.textview)
+        self.textview.show()
+
+        app.window.get_child().pack_start(self.scrolled_window, False, True, 0)
+        app.window.get_child().reorder_child(self.scrolled_window, 2)
+
+        self.textview.set_name("syncLyricsWindow")
+        self.styleLyricsWindow()
+
+        self.adjustment.set_value(0)
+
+        self.scrolled_window.show()
+
+        self._syncTimer = GLib.timeout_add(self.SYNC_PERIOD, self._sync)
+        self.buildData()
+        self.timerControl()
+
+    def styleLyricsWindow(self):
+        qltk.add_css(self.textview, """
+            #syncLyricsWindow {{
+                background-color: {0};
+                color: {1};
+                font-size: {2}px;
+                font-weight: bold;
+            }}
+        """.format(self.getBgColor(), self.getTextColor(), self.getFontSize()))
+
+    def curPosition(self):
+        return app.player.get_position()
+
+    def disabled(self):
+        self.clearTimers()
+        GLib.source_remove(self._syncTimer)
+        self.textview.destroy()
+        self.scrolled_window.destroy()
+
+    def buildData(self):
+        self.textbuffer.set_text("")
+        if app.player.song is not None:
+            #check in same location as track
+            trackName = app.player.song.get("~filename")
+            newLrc = os.path.splitext(trackName)[0] + ".lrc"
+            if self._currentLrc != newLrc:
+                self._lines = []
+                if os.path.exists(newLrc):
+                    self.parseLrcFile(newLrc)
+            self._currentLrc = newLrc
+
+    def parseLrcFile(self, filename):
+        rawFile = ""
+        with open(filename, 'r') as lrcfile:
+            rawFile = lrcfile.read()
+
+        rawFile = rawFile.replace("\n", "")
+        begin = 0
+        keepReading = len(rawFile) != 0
+        tmp_dict = {}
+        compressed = []
+        while keepReading:
+            lyricsLine = ""
+            nextFind = rawFile.find("[", begin + 1)
+            if(nextFind == -1):
+                keepReading = False
+                lyricsLine = rawFile[begin:]
+            else:
+                lyricsLine = rawFile[begin:nextFind]
+            begin = nextFind
+
+            #parse lyricsLine
+            if not lyricsLine[1].isdigit():
+                continue
+            closeBracket = lyricsLine.find("]")
+            timeObject = datetime.strptime(lyricsLine[1:closeBracket],
+                                           '%M:%S.%f')
+            timeStamp = (timeObject.minute * 60000 + timeObject.second * 1000
+                         + timeObject.microsecond / 1000)
+            words = lyricsLine[closeBracket + 1:]
+            if words == "":
```

Comment:
Is `if not words`, or better, `if words` with reversed logic sufficient here? 

If not, depending on the difference between various _falsey_ things (empty string, `None`, `False` etc) is risky.